tests/jobs/test_scheduler_job.py

```diff
@@ -2100,14 +2105,27 @@ def test_dagrun_timeout_verify_max_active_runs(self):
         assert isinstance(orm_dag.next_dagrun, datetime.datetime)
         assert isinstance(orm_dag.next_dagrun_create_after, datetime.datetime)
 
-        # TODO[HA] Verify dag failure callback request sent to file processor
+        # Verify dag failure callback request is added to dagrun.callback
+        assert dr.callback == DagCallbackRequest(
+            full_filepath=dr.dag.fileloc,
+            dag_id=dr.dag_id,
+            is_failure_callback=True,
+            execution_date=dr.execution_date,
+            msg="timed_out"
+        )
+
+        # Verify dag failure callback request is sent to file processor
+        scheduler.processor_agent = mock.Mock()
+        scheduler.processor_agent.send_callback_to_execute = mock.Mock()
+        scheduler._send_dag_callbacks_to_processor(dr)
```

Comment:
I think this method should be called from within `scheduler._scheduler_dag_run` -- this test looks a bit "artificial" otherwise (i.e. we're having to replicate a the call sequence from SchedulerJob here, which means things could change in SchedulerJob and these tests could still pass.

Reply:
Fixed in https://github.com/astronomer/airflow/pull/913/commits/1af9b903b122cd7f71bfd1f27bb719e0f00c755f